cpp/src/io/csv/csv_writer.cu

```diff
@@ -0,0 +1,284 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+     * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file csv_writer.cu  code to create csv file
+ *
+ * CSV Writer
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <cuda_runtime.h>
+#include <nvstrings/NVStrings.h>
+#include <rmm/rmm.h>
+#include <rmm/thrust_rmm_allocator.h>
+#include "cudf.h"
+#include "utilities/error_utils.hpp"
+
+
+// called by write_csv method
+NVStrings* column_to_strings_csv( gdf_column* col, const char* delimiter, const char* strue, const char* sfalse )
+{
+    NVStrings* rtn = nullptr;
+    unsigned int rows = (unsigned int)col->size;
+    unsigned char* valid = (unsigned char*)col->valid;
+    switch( col->dtype )
+    {
+        case GDF_STRING:
+            rtn = ((NVStrings*)col->data)->copy();
+            break;
+        case GDF_INT8:
+            rtn = NVStrings::create_from_bools((const bool*)col->data,rows,strue,sfalse,valid);
+            break;
+        case GDF_INT32:
+            rtn = NVStrings::itos((const int*)col->data,rows,valid);
+            break;
+        case GDF_INT64:
+            rtn = NVStrings::ltos((const long*)col->data,rows,valid);
```

Comment:
Use fixed-width types instead of machine dependent types, i.e, `int64_t` vs `long`.